src/lumigo_tracer/lambda_tracer/lambda_reporter.py

```diff
@@ -275,15 +289,84 @@ def _get_prioritized_spans(
             span_metadata_size = spans_to_send_sizes[index]
             span_size = get_event_base64_size(span)
 
-            if current_size + span_size - span_metadata_size < request_max_size:
+            if current_size + span_size - span_metadata_size < buffered_max_size:
                 spans_to_send_dict[index] = span
                 current_size += span_size - span_metadata_size
             else:
                 # This is an optimization step. If the spans are too big, don't try to send them.
                 too_big_spans += 1
                 if too_big_spans >= too_big_spans_threshold:
                     break
-    return list(spans_to_send_dict.values())
+
+        # If we dropped spans we need to update the enrichment spans dropped spans reasons
+        final_spans_list = list(spans_to_send_dict.values())
+        if len(spans_to_send_dict) != len(msgs):
+            with lumigo_safe_execute(
+                "create_request_body: smart span selection: updating enrichment span"
+            ):
+                final_spans_list = _update_enrichment_span_about_prioritized_spans(
+                    spans_to_send_dict, msgs, current_size, request_max_size
+                )
+
+    return final_spans_list
+
+
+def _update_enrichment_span_about_prioritized_spans(
+    spans_dict: Dict[int, Dict[str, Any]],
+    msgs: List[Dict[str, Any]],
+    current_size: int,
+    max_size: int,
+) -> List[Dict[str, Any]]:
+    """
+    Looks at the given spans about to be sent + the total number of messages,
+    and updates the enrichment spans about any dropped spans
+    @param spans_dict: The mapping of spans about to be sent
+    @param msgs: The complete list of spans created
+    @param current_size: The current size of all the spans in spans_dict
+    @param max_size: The maximum size of all spans together
+    @return: An updated list of spans, including the updated enrichment span
+    """
+    # Split spans into enrichment span and all other spans
+    enrichment_spans = []
+    spans = []
+    spans_size = current_size
+    for span in spans_dict.values():
+        if span.get("type") == ENRICHMENT_TYPE:
+            enrichment_spans.append(span)
+            current_span_size = get_event_base64_size(span)
+            spans_size -= current_span_size
+        else:
+            spans.append(span)
+
+    if not not enrichment_spans or len(enrichment_spans) > 1:
+        # We should never get here, if we did it probably means a bug in the tracer code.
+        get_logger().warning(f"Got unsupported number of enrichment spans: {len(enrichment_spans)}")
+        return list(spans_dict.values())
+
+    dropped_spans_due_to_size_limit = len(msgs) - len(spans_dict)
+    if not dropped_spans_due_to_size_limit:
```

Comment:
I think its more clear to write `dropped_spans_due_to_size_limit == 0`

Reply:
Done